### PR TITLE
Add listenFields function

### PR DESCRIPTION
**Pull request by Epimodev**
In some forms, we might need to apply some side effects when a field changed for example:
- display/hide fields depending on the value of a previous field
- update/reset the value of a field when we change another one

So here is a suggestion with a `listenFields` function returned by `useForm`:

```tsx
const { listenFields } = useForm({/* form config */})

React.useEffect(() => {
  const removeListener = listenFields(["firstName", "lastName"], ({ firstName, lastName }) => {
    /* ... */
  });

  return () => {
    removeListener();
  };
}, []);
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -194,6 +194,34 @@ A component that listen for fields states changes. Useful when a part of your co
 </FieldsListener>
 ```
 
+#### listenField
```
Comment (zoontek) [inline README.md:197]: You forgot to update the documentation (`listenField` -> `listenFields`)

File: src/index.ts [hunk 2]
```diff
@@ -34,6 +34,13 @@ export type FieldState<Value, ErrorMessage = string> = {
   error: ErrorMessage | undefined;
 };
 
+type ListenFieldsFunction<Values extends Record<string, unknown>, ErrorMessage = string> = <
```
Comment (zoontek) [inline src/index.ts:37]: There's no need to extract the type, you can inline it inside the `Form` type, like other `api` methods

File: src/index.ts [hunk 3]
```diff
@@ -373,6 +382,27 @@ export const useForm = <Values extends Record<string, unknown>, ErrorMessage = s
       return Promise.resolve(internalValidateField(name));
     };
 
+    const listenFields: ListenFieldsFunction<Values, ErrorMessage> = (names, onUpdate) => {
```
Comment (zoontek) [inline src/index.ts:385]: I would rename `onUpdate` -> `listener`

Comment (zoontek) [reply src/index.ts:385]: You can use `Contract["listenFields"]`, like the other methods

File: src/index.ts [hunk 4]
```diff
@@ -373,6 +382,27 @@ export const useForm = <Values extends Record<string, unknown>, ErrorMessage = s
...
+      const callback = () => {
```
Comment (zoontek) [inline src/index.ts:386]: No need for variable declaration + you are in the `api` scope, no need for `api.xxx`

```ts
    const listenFields: Contract["listenFields"] = (names, listener) => {
      const callback = () => {
        listener(
          names.reduce((acc, name) => {
            acc[name] = transformState(name, states.current[name], { sanitize: false });
            return acc;
          }, {} as { [N1 in typeof names[number]]: FieldState<Values[N1], ErrorMessage> })
        );
      };

      names.forEach((name) => callbacks.current[name].add(callback));

      return () => {
        names.forEach((name) => callbacks.current[name].delete(callback));
      };
    };
```

File: README.md (hunk 1 above)
Comment (Epimodev) [reply README.md:197]: oups 😅

File: src/index.ts (hunk 4 above)
Comment (Epimodev) [reply src/index.ts:386]: I like create variables, it helps me for debugging, but this way is good for me. I made the change

File: README.md (hunk 1 above)
Comment (zoontek) [reply README.md]: @Epimodev You clearly need holidays. I heard about an offsite… 😬

**Comment by Epimodev** [comment]
@zoontek thanks for those comments! I just pushed a commit with all request changes

File: README.md [hunk 5]
```diff
@@ -194,6 +194,34 @@ A component that listen for fields states changes. Useful when a part of your co
 </FieldsListener>
 ```
 
+#### listenFields
+
+A function that listen a field states changes. Useful when you want to apply side effects on value changed.
+
+```tsx
+React.useEffect(() => {
+  const removeListener = listenField(
```
Comment (zoontek) [inline README.md:203]: Well, this need to be updated too 😅

Comment (Epimodev) [reply README.md]: I think this time this is good

File: README.md [hunk 6]
```diff
@@ -196,24 +196,22 @@ A component that listen for fields states changes. Useful when a part of your co
 
 #### listenFields
 
-A function that listen a field states changes. Useful when you want to apply side effects on value changed.
+A function that listen fields states changes. Useful when you want to apply side effects on value changed.
```
Comment (zoontek) [inline README.md:199]: `on value changed` -> `on values change`

File: README.md (hunk 5 above)
Comment (zoontek) [reply README.md]: Close 😄

File: README.md (hunk 6 above)
Comment (Epimodev) [reply README.md]: ho! right 👍

**Comment by zoontek** [comment]
@Epimodev LGTM, great work 👍

**Comment by Epimodev** [comment]
thanks a lot 🙏

**Pull request merged**